libcloud/container/drivers/lxd.py

```diff
@@ -0,0 +1,1269 @@
+        response_dict = response.parse_body()
+        assert_response(response_dict=response_dict, status_code=200)
+        meta = response_dict["metadata"]
+        return LXDServerInfo.build_from_response(metadata=meta)
+
+    def deploy_container(self, name, image, cluster=None,
+                         parameters=None, start=True,
+                         architecture=default_architecture,
+                         profiles=[default_profiles],
+                         ephemeral=default_ephemeral,
+                         config=None, devices=None,
+                         instance_type=None):
+
+        """
+        Create a new container
+        Authentication: trusted
+        Operation: async
+        Return: background operation or standard error
+
+        :param name: The name of the new container.
+        64 chars max, ASCII, no slash, no colon and no comma
+        :type  name: ``str``
+
+        :param image: The container image to deploy. Currently not used
+        :type  image: :class:`.ContainerImage`
+
+        :param cluster: The cluster to deploy to, None is default
+        :type  cluster: :class:`.ContainerCluster`
+
+        :param parameters: Container Image parameters.
+        This parameter should represent the
+        the ``source`` dictioanry expected by the  LXD API call. For more
+        information how this parameter should be structured see
+        https://github.com/lxc/lxd/blob/master/doc/rest-api.md
+        :type  parameters: ``str``
+
+        :param start: Start the container on deployment. True is the default
+        :type  start: ``bool``
+
+        :param architecture: string e.g. x86_64
+        :type  architecture: ``str``
+
+        :param profiles: List of profiles
+        :type  profiles: ``list``
+
+        :param ephemeral: Whether to destroy the container on shutdown
+        :type  ephemeral: ``bool``
+
+        :param config: Config override e.g.  {"limits.cpu": "2"}
+        :type  config: ``dict``
+
+        :param devices: optional list of devices the container should have
+        :type  devices: ``dict``
+
+        :param instance_type: An optional instance type
+        to use as basis for limits e.g. "c2.micro"
+        :type  instance_type: ``str``
+
+        :rtype: :class:`.Container`
+        """
+
+        c_params = {"architecture": architecture, "profiles": profiles,
+                    "ephemeral": ephemeral, "config": config,
+                    "devices": devices, "instance_type": instance_type}
+
+        if parameters:
+            parameters = json.loads(parameters)
+
+        container = self._deploy_container_from_image(name=name, image=image,
+                                                      parameters=parameters,
+                                                      cont_params=c_params)
+
+        if start:
+            container.start()
+
+        return container
+
+    def get_container(self, id, get_ip_addr=True):
```

Comment:
Please prefix ``get_ip_addr`` with ``ex_`` since it's not part of the standard API.